### ISSUE TITLE
Lots of errors when opening aspnet/SignalR

### ISSUE THREAD

**Issue by davidfowl** [OmniSharp/omnisharp-roslyn#795]
I'm running on OSX.

I didn't want to open an uber issue but it seems like there are multiple things happening here. The repro steps are to clone https://github.com/aspnet/SignalR and run build.sh. After the build finishes, open vs code at the root (`code .`).

1. I get warnings telling me there are unresolved dependencies even though restore ran.
2. I get 100s of errors talking about ambiguous references. It seems as though the sample projects don't see the src projects within the same solution so packages are being resolved...

![image](https://cloud.githubusercontent.com/assets/95136/23789882/971b6578-0531-11e7-9321-cc9a813570d0.png)

I'll try to pick it apart into specific issues but I was caught by surprise because I thought things would just work.

**Comment by davidfowl** [comment]
@DustinCampbell am assuming nobody is looking into this issue. There's a fundamental problem with project references in msbuild within omnisharp. If nobody gets to this issue I'll try to take a look this weekend. I can't use VS code anymore on any project in ASP.NET because of it.

**Comment by DustinCampbell** [comment]
Sorry @davidfowl. I forgot to post back here after trying to repro a few weeks ago. Interestingly, I can repro if I run `./build.sh` but I can't repro if I just `dotnet restore SignalR.sln` with .NET Core SDK 1.0.1. I do get several messages about unresolved dependencies which I dismiss with <kbd>ESC</kbd>. However, I don't get any errors.

Here's what I did in the Terminal:

```
git clone https://github.com/aspnet/SignalR.git
cd SignalR/
dotnet restore SignalR.sln
code .
```

No errors anywhere. 😞 

![image](https://cloud.githubusercontent.com/assets/116161/24745387/6da8f608-1a6a-11e7-9c13-b961dbb882ab.png)

Hovering over the methods where your screenshot shows errors or pressing <kbd>F12</kbd> on them shows that they're indeed binding properly.

I _suspect_ this has to do with the fact that the SignalR is using .NET Core SDK 2.0.0-preview1-005418, which is beyond RTM. Today OmniSharp ships RTM versions of the various MSBuild SDKs and points to those at designtime. So, at designtime, an older version of, say, Microsoft.Net.Web.Sdk is being used than what `./build.sh` just ran against. That's just a guess though. I haven't dug in deeply enough to know for sure.

**Comment by DustinCampbell** [comment]
got a fix for the unresolved references warnings. It was an issue with version ranges. SignalR's build uses 2.0.0-* throughout.

**Comment by DustinCampbell** [comment]
Nah, it wasn't about SDKs. Project references were getting added twice due to some code that got lost in a refactoring a few months ago: https://github.com/OmniSharp/omnisharp-roslyn/commit/ef4b3838d39666ed5ca9780ea133bfdc08ff5844.

I've got a fix that should make SignalR work nicely for you.

**Comment by DustinCampbell** [comment]
If you don't want to wait for a C# for VS Code update (which'll be a little while yet), you can build the latest omnisharp-roslyn dev branch and set the following settings in VS Code:

* `"omnisharp.path"` : "<path-to-the-net46-OmniSharp.exe>"
* `"omnisharp.useMono"` : true

Build instructions are [here](https://github.com/OmniSharp/omnisharp-roslyn/blob/dev/BUILD.md)
